dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/utils/RateLimiterTest.java

```diff
@@ -0,0 +1,37 @@
+package com.salesforce.dockerfileimageupdate.utils;
+
+import org.testng.annotations.Test;
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.assertEquals;
+import java.time.Duration;
+import java.lang.InterruptedException;
+
+public class RateLimiterTest extends Thread{
+    
+    @Test
+    public void testConsume() throws InterruptedException{
+        Thread t1 = new Thread(){
+            @Override
+            public void run() {
+                try {
+                    RateLimiter rateLimiter = new RateLimiter(2,Duration.ofMinutes(2),Duration.ofMinutes(1));
+                    rateLimiter.consume();
+                    Thread t = Thread.currentThread();
+                    assertEquals(t.getState().toString(),"RUNNABLE");
+                    //thread will go in waiting state as the second token will be available only after 1 min
+                    rateLimiter.consume();
+                }catch(InterruptedException e){
+                    System.out.println("exception thrown");
+                }
+            }
+        };
+        t1.start();
+        Thread.sleep(30000);
```

Comment:
This is going to make this test really long. Let's shorten this sleep expectation.